x-pack/plugins/actions/server/builtin_action_types/slack.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import { IncomingWebhook } from '@slack/webhook';
+
+import { ActionType, ActionTypeExecutorOptions } from '../types';
+
+let IncomingWebhookImpl = IncomingWebhook;
+
+// for testing
+export function setIncomingWebhookImpl(incomingWebHook: any = IncomingWebhook): void {
+  IncomingWebhookImpl = incomingWebHook;
+}
+
+const CONFIG_SCHEMA = Joi.object().keys({
+  webhookUrl: Joi.string().required(),
+});
+
+const PARAMS_SCHEMA = Joi.object().keys({
+  message: Joi.string().required(),
+});
+
+export const actionType: ActionType = {
+  id: 'kibana.slack',
+  name: 'slack',
+  unencryptedAttributes: [],
+  validate: {
+    params: PARAMS_SCHEMA,
+    config: CONFIG_SCHEMA,
+  },
+  executor,
+};
+
+async function executor({ config, params, services }: ActionTypeExecutorOptions): Promise<any> {
+  const { webhookUrl } = config;
+  const { message } = params;
+
+  // TODO: do we need an agent for proxy access?
+  const webhook = new IncomingWebhookImpl(webhookUrl);
+
+  // TODO: should we have a standardize response for executor?
```

Comment:
Yeah, there's definitely no standard response at this time. Though, we can create a one. What you have right now is ok since the caller doesn't do anything with the response (yet). https://github.com/elastic/kibana/pull/37042/files#diff-505edb052f98e69dc9e16d579cdc068cR44

Reply:
++ to standardized response